scripts/init_db.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -ex # hanldling errors and Debuging purpose 
+# Declare environment variables
+
+DB_USER="${POSTGRES_USER:=postgres}"
+DB_PASSWORD="${POSTGRES_PASSWORD:=password}"
+DB_NAME="${POSTGRES_DB:=newsletter}"
+DB_PORT="${POSTGRES_PORT:=5432}"
+DB_HOST="${POSTGRES_HOST:=localhost}"
+SKIP_DOCKER=${SKIP_DOCKER:=false}
+DATABASE_URL="postgres://$DB_USER:$DB_PASSWORD@localhost:5432/$DB_DATABASE"
+
+# Check if Docker is installed
+if [ "$SKIP_DOCKER" = "false" ]; then
+    if ! docker --version > /dev/null 2>&1; then
+        echo "Docker required but not found"
+        exit 1
+    fi
+
+    echo "docker initialisation"
+        docker run --name "$DB_NAME"_container \
+        -e POSTGRES_USER="$DB_USER" \
+        -e POSTGRES_PASSWORD="$DB_PASSWORD" \
+        -e POSTGRES_DB="$DB_NAME" \
+        -p "$DB_PORT":5432 \
+        -d postgres
```

Comment:
You haven't addressed my comment about increasing the maximum number of connections that will be used for testing purposes.
```suggestion
        -d postgres
        postgres -N 1000
```